kong/plugins/ai-prompt-template/templater.lua

```diff
@@ -1,93 +1,57 @@
-local _S = {}
-
--- imports
-local fmt           = string.format
---
-
--- globals
-local GSUB_REPLACE_PATTERN = "{{([%w_]+)}}"
---
-
-local function backslash_replacement_function(c)
-  if c == "\n" then
-     return "\\n"
-  elseif c == "\r" then
-     return "\\r"
-  elseif c == "\t" then
-     return "\\t"
-  elseif c == "\b" then
-     return "\\b"
-  elseif c == "\f" then
-     return "\\f"
-  elseif c == '"' then
-     return '\\"'
-  elseif c == '\\' then
-     return '\\\\'
-  else
-     return string.format("\\u%04x", c:byte())
-  end
-end
+local cjson = require "cjson.safe"
 
-local chars_to_be_escaped_in_JSON_string
-= '['
-..    '"'    -- class sub-pattern to match a double quote
-..    '%\\'  -- class sub-pattern to match a backslash
-..    '%z'   -- class sub-pattern to match a null
-..    '\001' .. '-' .. '\031' -- class sub-pattern to match control characters
-.. ']'
-
--- borrowed from turbo-json
-local function sanitize_parameter(s)
-  if type(s) ~= "string" or s == "" then
-    return nil, nil, "only string arguments are supported"
-  end
+local _M = {}
 
-  -- check if someone is trying to inject JSON control characters to close the command
-  if s:sub(-1) == "," then
-    s = s:sub(1, -1)
-  end
 
-  return s:gsub(chars_to_be_escaped_in_JSON_string, backslash_replacement_function), nil
-end
 
-function _S:new(o)
-  local o = o or {}
-  setmetatable(o, self)
-  self.__index = self
+--- sanitize table of parameters.
+-- Any non-string values will be dropped.
+-- @tparam table params the kv table to sanitize, if not a table will be replace with an empty table
+-- @return the same table, with escaped values (without quotes)
+local function sanitize_parameters(params)
+  local result = {}
 
-  return o
+  if type(params) ~= "table" then
+    return result
```

Comment:
just checking: are we expecting this to happen? Should we return `nil, "some err"`?

Reply:
done